absl/absl.h

```diff
@@ -5,10 +5,41 @@
 
 namespace nda {
 
-// interval -> string.
+// interval -> string as the half-open interval [begin, end).
+//
+// Stringifies only the values, not whether they are static or dynamic.
 template <typename Sink, index_t Min_ = dynamic, index_t Extent_ = dynamic>
-void AbslStringify(Sink& sink, const nda::interval<Min_, Extent_>& p) {
-  absl::Format(&sink, "[%v, %v)", p.min(), p.min() + p.extent());
+void AbslStringify(Sink& sink, const interval<Min_, Extent_>& i) {
+  absl::Format(&sink, "[%v, %v)", i.min(), i.min() + i.extent());
+}
+
+// dim -> string as "dim(min, extent, stride)".
+//
+// Stringifies only the values, not whether they are static or dynamic.
+template <typename Sink, index_t Min_ = dynamic, index_t Extent_ = dynamic,
+    index_t Stride_ = dynamic>
+void AbslStringify(Sink& sink, const dim<Min_, Extent_, Stride_>& d) {
+  absl::Format(&sink, "dim(%v, %v, %v)", d.min(), d.extent(), d.stride());
+}
+
+// shape -> string as "shape<`rank`>(dims...).
+//
+// Stringifies only the rank and each dim's values, not whether they are static
+// or dynamic.
+//
+// TODO(jiawen): This can be implemented at compile-time using index_sequence
```

Comment:
I don't think this should be a TODO, we should just not do this at compile time. The code size impact would be annoying for something that isn't helpful for performance.

Reply:
👍